src/daemons/publishDaemon.ts

```diff
@@ -0,0 +1,135 @@
+import { AMQPMessage } from '@cloudamqp/amqp-client'
+import admin from 'firebase-admin'
+import { base64 } from 'rfc4648'
+
+import { getApiKeyByKey } from '../db/couchApiKeys'
+import {
+  DeviceRow,
+  getDeviceById,
+  getDevicesByLoginId
+} from '../db/couchDevices'
+import { DbConnections } from '../db/dbConnections'
+import { logger } from '../util/logger'
+import { asRabbitMessage, SendableMessage } from '../util/pushSender'
+import { runDaemon } from './runDaemon'
+
+runDaemon(async tools => {
+  const { connections } = tools
+
+  const consumer = await connections.queue.subscribe(
+    { noAck: false },
+    message => {
+      handleMessage(connections, message).catch(error => {
+        logger.warn({
+          msg: 'Failed to handle message',
+          body: message.bodyToString(),
+          error
+        })
+      })
+    }
+  )
+
+  // Will wait until we have an error, which closes the channel:
+  await consumer.wait()
```

Comment:
Will errors only occur if a subscriber callback throws, or are there other connection errors we should handle?